Compiled from "CWE617_Reachable_Assertion__length_14.java"
public class testcases.CWE617_Reachable_Assertion.CWE617_Reachable_Assertion__length_14 extends testcasesupport.AbstractTestCase {
  static final boolean $assertionsDisabled;

  public testcases.CWE617_Reachable_Assertion.CWE617_Reachable_Assertion__length_14();
    Code:
       0: aload_0
       1: invokespecial #1                  // Method testcasesupport/AbstractTestCase."<init>":()V
       4: return

  public void bad() throws java.lang.Throwable;
    Code:
       0: getstatic     #7                  // Field testcasesupport/IO.staticFive:I
       3: iconst_5
       4: if_icmpne     29
       7: getstatic     #13                 // Field $assertionsDisabled:Z
      10: ifne          29
      13: ldc           #19                 // String
      15: invokevirtual #21                 // Method java/lang/String.length:()I
      18: ifgt          29
      21: new           #27                 // class java/lang/AssertionError
      24: dup
      25: invokespecial #29                 // Method java/lang/AssertionError."<init>":()V
      28: athrow
      29: return

  private void good1() throws java.lang.Throwable;
    Code:
       0: getstatic     #7                  // Field testcasesupport/IO.staticFive:I
       3: iconst_5
       4: if_icmpeq     15
       7: ldc           #30                 // String Benign, fixed string
       9: invokestatic  #32                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      12: goto          37
      15: getstatic     #13                 // Field $assertionsDisabled:Z
      18: ifne          37
      21: ldc           #36                 // String cwe617
      23: invokevirtual #21                 // Method java/lang/String.length:()I
      26: ifgt          37
      29: new           #27                 // class java/lang/AssertionError
      32: dup
      33: invokespecial #29                 // Method java/lang/AssertionError."<init>":()V
      36: athrow
      37: return

  private void good2() throws java.lang.Throwable;
    Code:
       0: getstatic     #7                  // Field testcasesupport/IO.staticFive:I
       3: iconst_5
       4: if_icmpne     29
       7: getstatic     #13                 // Field $assertionsDisabled:Z
      10: ifne          29
      13: ldc           #36                 // String cwe617
      15: invokevirtual #21                 // Method java/lang/String.length:()I
      18: ifgt          29
      21: new           #27                 // class java/lang/AssertionError
      24: dup
      25: invokespecial #29                 // Method java/lang/AssertionError."<init>":()V
      28: athrow
      29: return

  public void good() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: invokevirtual #38                 // Method good1:()V
       4: aload_0
       5: invokevirtual #41                 // Method good2:()V
       8: return

  public static void main(java.lang.String[]) throws java.lang.ClassNotFoundException, java.lang.InstantiationException, java.lang.IllegalAccessException;
    Code:
       0: aload_0
       1: invokestatic  #44                 // Method mainFromParent:([Ljava/lang/String;)V
       4: return

  static {};
    Code:
       0: ldc           #14                 // class testcases/CWE617_Reachable_Assertion/CWE617_Reachable_Assertion__length_14
       2: invokevirtual #48                 // Method java/lang/Class.desiredAssertionStatus:()Z
       5: ifne          12
       8: iconst_1
       9: goto          13
      12: iconst_0
      13: putstatic     #13                 // Field $assertionsDisabled:Z
      16: return
}
